### PR TITLE
More fix warnings

### PR DESCRIPTION
**Pull request by ylipacbio**
Fix more warnings.
Move variables from global space to PacBio.
Change enum FileType to class enum FileType.

### CODE REVIEW

File: alignment/algorithms/alignment/ExtendAlign.hpp [hunk 1]
```diff
@@ -16,6 +16,11 @@
 //       No need to change data type of query pos to DNALength, 
 //       since it's unlikely to have > 2G bases per zmw.
 
+//FIXME: change data type of target pos from int to GenomeLength
```
Comment (pb-cdunn) [inline alignment/algorithms/alignment/ExtendAlign.hpp]: You can create a task for this in GH.

File: alignment/suffixarray/ssort.hpp [hunk 2]
```diff
@@ -256,15 +257,17 @@ ssort(SAIndex a[], SAIndex s[])
     SAIndex *q = 0;
 #	define al a
 #	define pl p
-#	define finish(r)  if(1){result=r; goto out;}else
+#	define finish(r)  if(1){result=r; goto out;}
```
Comment (pb-cdunn) [inline alignment/suffixarray/ssort.hpp]: Super-dangerous macro! The reason for the trailing `else` was to ensure that  a dangling `else` would not accidentally be attached to *this* `if`. Very important! Without it, you have to be very careful of all the places where `finish()` is used.

But a better way to write the macro is

    #	define finish(r)  do {result=r; goto out;} while(0)

a common old-school trick. There are no major gotchas with that version.

File: alignment/suffixarray/ssort.hpp [hunk 3]
```diff
@@ -296,7 +299,8 @@ ssort(SAIndex a[], SAIndex s[])
                     finish(2);
         }
 
-        for(i=n; --k>=0; ) {		/* (2) order */
+        //for(i=n; --k>=0; ) {		/* (2) order */
+        for(i=n; k>=1 && --k; ) {		/* (2) order */
```
Comment (pb-cdunn) [inline alignment/suffixarray/ssort.hpp]: I think this is a new bug. The problem is `&&`. When `k` goes from `1` to `0`, that will evaluate to `false` (`true && 0`), whereas the old code would quit when `k` goes to `-1`.

Maybe I'm wrong, but obviously this is not clear. (And btw, if `k` starts at `0`, there is a big problem in the old code. Is that ever possible?) Please change. Maybe a simple while-loop, with initial conditions before it and the decrement at the top of the loop block.

File: alignment/tuples/DNATuple.hpp [hunk 4]
```diff
@@ -19,30 +19,41 @@ class DNATuple : public BaseTuple {
 public:
     DNALength pos;
 
+    DNATuple();
```
Comment (pb-cdunn) [inline alignment/tuples/DNATuple.hpp]: I don't understand this. Isn't this ctor automatically provided?

File: alignment/tuples/DNATuple.hpp [hunk 5]
```diff
@@ -19,30 +19,41 @@ class DNATuple : public BaseTuple {
...
+
+    DNATuple(const DNATuple & rhs);
```
Comment (pb-cdunn) [inline alignment/tuples/DNATuple.hpp]: This one too.

File: alignment/algorithms/alignment/ExtendAlign.hpp (hunk 1 above)
Comment (ylipacbio) [inline alignment/algorithms/alignment/ExtendAlign.hpp]: Haha, only if we have to support > 4G target sequence.

File: alignment/suffixarray/ssort.hpp (hunk 3 above)
Comment (ylipacbio) [inline alignment/suffixarray/ssort.hpp]: You are right, let me switch it back.

**Pull request merged**